actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -82,11 +81,19 @@ def format(formatter, filter = nil)
 
       private
 
-      def filter_routes(filter)
-        if filter
+      def filter_routes(options)
```

Comment:
Sorry for keeping this in limbo, but I don't think we should have both `filter` and `pattern`. In fact I think there's a way to get rid of `pattern`.

How about having the option parser do:

```ruby
opts.on("-c", "--controller [PATTERN]") do |pattern|
  options[: defaults_filter] = { controller: pattern.downcase.sub(/_?controller\z/, '') }
end

opts.on("-g", "--grep [PATTERN]") do |pattern|
  options[:defaults_filter] = ActionDispatch::Routing::ConsoleFormatter.defaults_filter_for(/#{pattern}/)
end
```

Later:

```ruby
if defaults_filter
  @routes.select do |route|
    defaults_filter.any? { |default, filter| route.defaults[default] =~ filter }
  end
else
  @routes
end
```

`defaults_filter_for` would just populate a hash where keys are defaults for a route, and the value is the pattern.

Make sense? :smile: